### PR TITLE
getallcontacts API

### PR DESCRIPTION
**Pull request by ahmadshaqqou**

### CODE REVIEW

File: api/getAllContact.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    } catch (PDOException $exception) {
+        http_response_code(500);
+        echo json_encode(["message" => "Database connection error: " . $exception->getMessage()]);
+        die(json_encode(["message" => "Database connection error: " . $exception->getMessage()]));
+    }
+}
+
+$conn = getConnection();
+
+$query = "SELECT * FROM Contact";
```
Comment (jfbarbery) [inline api/getAllContact.php:27]: Should we fetch all contacts in the databse, or specify by ID? IIRC we designed this API endpoint to filter by user ID

Comment (vphatfla) [inline api/getAllContact.php:27]: We don't need this API. What we need is a API that can getAllContacts of a particular user with user_id 
Imagine what happens if this API is not properly protected, hackers can get all information in the database without even trying. 
We need a parameter for this API as a user_id, the Query should look something like this 

```
"SELECT * FROM Contact WHERE user_id = ?;"
```

**Review by vphatfla — COMMENTED** [review]
Param needed

**Comment by vphatfla** [comment]
@ahmadshaqqou   please review this code and let me know if you want to use this for the api instead 

file: getAllContactsByUserId.php
```
<?php
// Load environment variables
require_once 'vendor/autoload.php';
$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->load();

// Set headers for JSON response
header("Content-Type: application/json; charset=UTF-8");
header("Access-Control-Allow-Methods: GET");

// Function to connect to the database
function getConnection() {
    $host = $_ENV['DB_HOST'];  // Fetch from .env
    $db_name = $_ENV['DB_NAME']; // Fetch from .env
    $username = $_ENV['DB_USER']; // Fetch from .env
    $password = $_ENV['DB_PASS']; // Fetch from .env

    try {
        $conn = new PDO("mysql:host=$host;dbname=$db_name;charset=utf8mb4", $username, $password);
        $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
        return $conn;
    } catch (PDOException $exception) {
        http_response_code(500);
        echo json_encode(["message" => "Database connection error: " . $exception->getMessage()]);
        die();
    }
}

// Check if the uid parameter is provided
if (!empty($_GET['uid'])) {
    $uid = intval($_GET['uid']);

    // Prepare database connection
    $conn = getConnection();

    // Prepare SQL query to fetch contacts based on uid
    $query = "SELECT * FROM Contact WHERE uid = :uid";
    $stmt = $conn->prepare($query);
    $stmt->bindParam(":uid", $uid);

    // Execute the query
    if ($stmt->execute()) {
        $contacts = $stmt->fetchAll(PDO::FETCH_ASSOC);

        // Check if any contacts were found
        if ($contacts) {
            // Respond with contacts array
            http_response_code(200);
            echo json_encode($contacts);
        } else {
            // No contacts found for the uid
            http_response_code(404);
            echo json_encode(["message" => "No contacts found for the provided uid"]);
        }
    } else {
        // Query execution failed
        http_response_code(500);
        echo json_encode(["message" => "Unable to execute query"]);
    }
} else {
    // uid parameter is missing
    http_response_code(400);
    echo json_encode(["message" => "uid parameter is missing"]);
}
?>

```

File: api/getAllContact.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?php
+// Set headers for JSON response
+header("Content-Type: application/json; charset=UTF-8");
+header("Access-Control-Allow-Methods: GET");
+
```
Comment (vphatfla) [inline api/getAllContact.php:5]: you can safely delete this file

File: api/getAllContactsByUserId.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
+<?php
+// Load environment variables
+require_once 'vendor/autoload.php';
+$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
+$dotenv->load();
+
```
Comment (vphatfla) [inline api/getAllContactsByUserId.php:6]: where is the http header?

Comment (ahmadshaqqou) [reply api/getAllContactsByUserId.php:11]: deleted them on accident should be fixed now

**Review by vphatfla — APPROVED** [review]

**Pull request merged**